src/commonMain/kotlin/exchange.dydx.abacus/output/input/AdjustIsolatedMarginInput.kt

```diff
@@ -0,0 +1,175 @@
+) {
+    companion object {
+        internal fun create(
+            existing: AdjustIsolatedMarginInputOptions?,
+            parser: ParserProtocol,
+            data: Map<*, *>?
+        ): AdjustIsolatedMarginInputOptions? {
+            Logger.d { "creating Adjust Isolated Margin Input Options\n" }
+
+            data?.let {
+                val needsSize = parser.asBool(data["needsSize"])
+
+                return if (existing?.needsSize != needsSize
+                ) {
+                    AdjustIsolatedMarginInputOptions(needsSize)
+                } else {
+                    existing
+                }
+            }
+            Logger.d { "Adjust Isolated Margin Input Options not valid" }
+            return null
+        }
+    }
+}
+
+@JsExport
+@Serializable
+data class AdjustIsolatedMarginInputSummary(
+    val crossFreeCollateral: Double?,
+    val crossMarginUsage: Double?,
+    val positionMargin: Double?,
+    val positionLeverage: Double?,
+    val liquidationPrice: Double?,
+) {
+    companion object {
+        internal fun create(
+            existing: AdjustIsolatedMarginInputSummary?,
+            parser: ParserProtocol,
+            data: Map<*, *>?
+        ): AdjustIsolatedMarginInputSummary? {
+            Logger.d { "creating Adjust Isolated Margin Input Summary\n" }
+
+            data?.let {
+                val crossFreeCollateral = parser.asDouble(data["crossFreeCollateral"])
+                val crossMarginUsage = parser.asDouble(data["crossMarginUsage"])
+                val positionMargin = parser.asDouble(data["positionMargin"])
+                val positionLeverage = parser.asDouble(data["positionLeverage"])
+                val liquidationPrice = parser.asDouble(data["liquidationPrice"])
+
+                return if (
+                    existing?.crossFreeCollateral != crossFreeCollateral ||
+                    existing?.crossMarginUsage != crossMarginUsage ||
+                    existing?.positionMargin != positionMargin ||
+                    existing?.positionLeverage != positionLeverage ||
+                    existing?.liquidationPrice != liquidationPrice
+                ) {
+                    AdjustIsolatedMarginInputSummary(
+                        crossFreeCollateral,
+                        crossMarginUsage,
+                        positionMargin,
+                        positionLeverage,
+                        liquidationPrice,
+                    )
+                } else {
+                    existing
+                }
+            }
+            Logger.d { "Adjust Isolated Margin Input Summary not valid" }
+            return null
+        }
+    }
+}
+
+@JsExport
+@Serializable
+enum class IsolatedMarginAdjustmentType(val rawValue: String) {
+    Add("ADD"),
+    Remove("REMOVE");
+
+    companion object {
+        operator fun invoke(rawValue: String) =
+            IsolatedMarginAdjustmentType.values().firstOrNull { it.rawValue == rawValue }
```

Comment:
Can use built-in Kotlin conversion here:
```
enum class IsolatedMarginAdjustmentType {
  ADD,
  REMOVE,
}
```

When parsing: `IsolatedMarginAdjustmentType.valueOf( ... )`

(also don't need semi-colons in kotlin :) )

Reply:
I think we move to the built-in kotlin conversion we lose access to `rawValue` (which is heavily leveraged in the FE code)

Reply:
oh the equivalent built-in is `.name`: https://kotlinlang.org/api/latest/jvm/stdlib/kotlin/-enum/name.html

Reply:
Do you happen to know what happens if we try to invoke an enum from a string and it doesn't exist? The current `invoke` method will result in `null`.

i.e. Protocol/Indexer adds `TRAILING_STOP` orders but it isn't defined in our `OrderType` enums. Will invoking the enum on that string result in null?

Reply:
Not sure if every platform will handle this the same way as well.

Reply:
ohhh that's a good callout.

Trying this out in Kotlin Playground shows us that this will throw an error: 
https://pl.kotl.in/XH_R97P6q

I also compiled an example Kotlin snippet to JS:
```
// Kotlin
enum class TestEnum {
  ADD,
  REMOVE,
}
```
```
// compiled JS
var TestEnum_ADD_instance;
var TestEnum_REMOVE_instance;
function valueOf(value) {
  switch (value) {
    case 'ADD':
      return TestEnum_ADD_getInstance();
    case 'REMOVE':
      return TestEnum_REMOVE_getInstance();
    default:
      TestEnum_initEntries();
      THROW_IAE('No enum constant value.');
      break;
  }
}
```
So can see where it calls `THROW_IAE`

Reply:
Yeah, mk seems like the current approach is the best we can do. I tried to explore if we could write a generic helper function to avoid writing this for all of our enums, but seems like it's not possible:
https://youtrack.jetbrains.com/issue/KT-42619/Add-a-static-valueOfOrNull-method-to-Enums-similar-to-valueOf